snakemake/rules.py

```diff
@@ -681,6 +681,15 @@ def apply_input_function(
         except IncompleteCheckpointException as e:
             value = incomplete_checkpoint_func(e)
             incomplete = True
+        except FileNotFoundError as e:
+            # Resources can depend on input files. Since expansion can happen during dryrun,
+            # where input files are not yet present, we need to skip such resources and
+            # mark them as [TBD].
+            if e.filename in aux_params["input"]:
+                # use zero for resource if it cannot yet be determined
+                value = TBDInt(0)
```

Comment:
This should then probably be changed into simply a TBD object instead of an Int. Otherwise, it is a big misleading.